### PR TITLE
Handle the gen_server timeout from eredis.

### PR DESCRIPTION
**Pull request by NikolaBorisov**
In which case we remap the cluster.

### CODE REVIEW

File: src/ecredis.erl [hunk 1]
```diff
@@ -385,6 +385,10 @@ get_successes_and_retries(#query{response = {error, <<"TRYAGAIN">>}, retries = R
     {[], [Query#query{retries = Retries + 1}]};
 get_successes_and_retries(#query{response = {error, no_connection}} = Query) ->
     handle_error(no_connection, Query);
+get_successes_and_retries(#query{response = {error, gen_server_timeout}} = Query) ->
```
Comment (vipin211) [inline src/ecredis.erl:388]: I think no_connection and gen_server_timeout should be handled in a similar fashion. Let me know what you think.

Comment (balusu-murali) [reply src/ecredis.erl:388]: agree - what do we do now on no-connection?

File: src/ecredis.erl [hunk 2]
```diff
@@ -577,6 +581,18 @@ group_by_destination(#query{command = Command,
     end.
 
 
+try_eredis_query(Pid, Command) ->
+    try
+        eredis_query(Pid, Command)
+    catch
+        exit : {timeout, St} ->
+            ?ERROR("eredis genserver timeout Pid: ~p Commands: ~p ~p", [Pid, Command, St]),
+            {error, gen_server_timeout};
```
Comment (balusu-murali) [inline src/ecredis.erl:590]: we should call it just timeout?

Comment (NikolaBorisov) [reply src/ecredis.erl:590]: Ok, lets call it just timeout. I thought it is interesting if it comes from the gen_server or maybe the socket itself.

File: src/ecredis.erl (hunk 1 above)
Comment (NikolaBorisov) [reply src/ecredis.erl:388]: Basically they are handled the same way.

Comment (NikolaBorisov) [reply src/ecredis.erl]: We remap and try to find again where we should send the query

**Review by vipin211 — APPROVED** [review]

**Review by mdonohue — APPROVED** [review]

**Review by balusu-murali — APPROVED** [review]

**Pull request merged**